daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -298,7 +298,9 @@ class ProcessorFactory private[japi] (private var pf: SProcessorFactory)
    */
   def onPath(path: String) = {
     val dp = pf.onPath(path).asInstanceOf[SDataProcessor]
-    new DataProcessor(dp)
+    val res = new DataProcessor(dp)
+    res.isError // ensure all errors have been detected before we return the DP
```

Comment:
Is there a reason we don't need to handle any errors detected here?  Do we assume whoever calls onPath handles the error?

Reply:
Yes, the idea isn't to handle the errors here as the API user may have their own ideas about when/how to do that. Ex: they may just have an `assert(!pf.isError)` and just fail out if there are errors because they are expecting to use well-known-good schemas only. 

The purpose of this is just to force all the lazy evaluation to happen before we return so that we can time things meaningfully. There's no advantage to the lazy evaluation continuing past this point, so we force it all.

Reply:
Makes sense.